tools/src/main/java/org/apache/kafka/tools/ClientCompatibilityTest.java

```diff
@@ -269,35 +269,21 @@ void testAdminClient() throws Throwable {
                         throw e.getCause();
                     }
                 },
-                () -> {
-                    while (true) {
-                        try {
-                            client.describeTopics(Collections.singleton("newtopic")).all().get();
-                            break;
-                        } catch (ExecutionException e) {
-                            if (e.getCause() instanceof UnknownTopicOrPartitionException)
-                                continue;
-                            throw e;
-                        }
-                    }
-                });
+                () ->  new CreateTopicsResultTester(client, Collections.singleton("newtopic")).invoke()
```

Comment:
Why do we need a class instead of a method here?

Reply:
It was throwing `NPathComplexity` style error due to nested conditional statements,

One of that error is,
> /kafka/tools/src/main/java/org/apache/kafka/tools/ClientCompatibilityTest.java:247:5: NPath Complexity is 3,405 (max allowed is 500). [NPathComplexity]

So, I moved the block of code to new method/class instead of supressing the error.

Reply:
I guessed the npathcomplexity thing. :) My question is why we have a class instead of just a method.